File: Binary_Search_Tree/Server/src/tree/Node.java
package tree;

/**
 * Node of tree/
 * @param <T> type of elements
 * @author Paweł Data
 */
public class Node<T extends Comparable<T>> {
    private T value;
    private Node<T> left;
    private Node<T> right;

    /**
     * Class constructor.
     */
    public Node() {}

    /**
     * Class constructor.
     * @param value node's value
     */
    public Node(T value) {
        this.value = value;
    }

    /**
     * Get value.
     * @return value
     */
    public T getValue() {
        return this.value;
    }

    /**
     * Set value.
     * @param value value
     */
    public void setValue(T value) {
        this.value = value;
    }

    /**
     * Set left node.
     * @param node left node
     */
    public void setLeftNode(Node<T> node) {
        this.left = node;
    }

    /**
     * Get left node.
     * @return left node
     */
    public Node<T> getLeftNode() {
        return this.left;
    }

    /**
     * Set right node.
     * @param node right node
     */
    public void setRightNode(Node<T> node) {
        this.right = node;
    }

    /**
     * Get right node.
     * @return right node
     */
    public Node<T> getRightNode() {
        return this.right;
    }
}

File: Binary_Search_Tree_cpp/src/Node.cpp
#include<iostream>

/**
 * Node of tree/
 * @param <T> type of elements
 * @author Paweł Data
 */
template<typename T> class Node {
private:
    T value;
    Node<T>* left;
    Node<T>* right;

public:

    /**
     * Class constructor.
     */
    Node(T v) {
        value = v;
        left = NULL;
        right = NULL;
    }

    T getValue() {
        return value;
    }

    /**
     * Set value.
     * @param value value
     */
    void setValue(T v) {
        value = v;
    }

    /**
     * Set left node.
     * @param node left node
     */
    void setLeftNode(Node<T>* node) {
        left = node;
    }

    /**
     * Get left node.
     * @return left node
     */
    Node<T> getLeftNode() {
        return left;
    }

    /**
     * Set right node.
     * @param node right node
     */
    void setRightNode(Node<T>* node) {
        right = node;
    }

    /**
     * Get right node.
     * @return right node
     */
    Node<T> getRightNode() {
        return right;
    }
};